source/mesh_deformation/interface.cc

```diff
@@ -93,6 +97,24 @@ namespace aspect
     }
 
 
+    template <int dim>
+    void
+    MeshDeformationHandler<dim>::initialize ()
+    {
+      // Get pointer to initial topography model
+      topo_model = const_cast<InitialTopographyModel::Interface<dim>*>(&this->get_initial_topography_model());
```

Comment:
Ah, that's really unfriendly. Why do you need to cast away the `const`ness? This suggests some kind of design failure.